### ISSUE TITLE
Add categorical cross-entropy derivative

### ISSUE THREAD

**Issue by opixelum** [opixelum/brain-disk#1]
The categorical cross-entropy derivative is used for backpropagation when we use softmax, since the softmax derivative can't be used for backpropagation.
